### PR TITLE
Secure Update password

### PR DESCRIPTION
**Pull request by nikpawar89**
User now has to authenticate the update password request by password.

### CODE REVIEW

File: fineract-provider/src/main/java/org/apache/fineract/useradministration/api/UsersApiResource.java [hunk 1]
```diff
@@ -140,30 +148,45 @@ public String create(final String apiRequestBodyAsJson) {
...
-        final CommandWrapper commandRequest = new CommandWrapperBuilder() //
-                .updateUser(userId) //
-                .withJson(apiRequestBodyAsJson) //
-                .build();
-
-        final CommandProcessingResult result = this.commandsSourceWritePlatformService.logCommandSource(commandRequest);
-
-        return this.toApiJsonSerializer.serialize(result);
+  @PUT
+  @Path("/{userId}/usrnme/{usrnme}/currentPass/{currentPass}")
```
Comment (rajuan) [inline fineract-provider/src/main/java/org/apache/fineract/useradministration/api/UsersApiResource.java:152]: 1. update API is used for all types of updates including firstname..... we cannot expect that a firstname update by back office user would need him to know the password of the user. This change cannot be accepted.
2. If the expectation is that an existing user is trying to modify the password of their own userId, then basic/oauth authentication itself would have validated the current password.... so no need to check again

**Comment by nikpawar89** [comment]
After implementing this code, I did few tests to change the user details. Nothing accept 'updatepassword case' was asked password. For now I have just proposed the change, there are still vulnerabilities in this approach which I discovered  later. I am going to Fix them and would come back later. Thanks for the feedback

File: fineract-provider/src/main/java/org/apache/fineract/useradministration/api/UsersApiResource.java (hunk 1 above)
Comment (nikpawar89) [inline fineract-provider/src/main/java/org/apache/fineract/useradministration/api/UsersApiResource.java:152]: After implementing this code, I did few tests to change the user details. Nothing accept 'updatepassword case' was asked password. For now I have just proposed the change, there are still vulnerabilities in this approach which I discovered later. I am going to Fix them and would come back later. Thanks for the feedback

Comment (MexinaD) [reply fineract-provider/src/main/java/org/apache/fineract/useradministration/api/UsersApiResource.java:152]: Hell @nikpawar  did you work on this?

**Comment by vorburger** [comment]
https://github.com/apache/fineract/pull/639/

**Pull request closed without merging**